otel-allocator/allocation/allocator.go

```diff
@@ -0,0 +1,130 @@
+package allocation
+
+import (
+	"log"
+	"sync"
+)
+
+/*
+	Load balancer will serve on an HTTP server exposing /jobs/<job_id>/targets <- these are configured using least connection
+	Load balancer will need information about the collectors in order to set the URLs
+	Keep a Map of what each collector currently holds and update it based on new scrape target updates
+*/
+
+// Create a struct that holds collector - and jobs for that collector
+// This struct will be parsed into endpoint with collector and jobs info
+
+type collector struct {
+	Name       string
+	NumTargets int
+}
+
+// allocator makes decisions to distribute work among
+// a number of OpenTelemetry collectors based on the number of targets.
+// Users need to call SetTargets when they have new targets in their
+// clusters and call Reshard to process the new targets and reshard.
+
+type Allocator struct {
+	m     sync.Mutex
+	cache displayCache
+
+	targetsWaiting map[string]TargetItem // temp buffer to keep targets that are waiting to be processed
+
+	collectors map[string]*collector // all current collectors
+
+	nextCollector *collector
+	targetItems   map[string]*TargetItem
+}
+
+// findNextCollector finds the next collector with less number of targets.
+func (allocator *Allocator) findNextCollector() {
+	for _, v := range allocator.collectors {
+		if v.NumTargets < allocator.nextCollector.NumTargets {
+			allocator.nextCollector = v
+		}
+	}
+}
+
+// SetTargets accepts the a list of targets that will be used to make
+// load balancing decisions. This method should be called when where are
+// new targets discovered or existing targets are shutdown.
+func (allocator *Allocator) SetTargets(targets []TargetItem) {
+	// Dump old data
+	allocator.m.Lock()
+	for k := range allocator.targetsWaiting {
+		delete(allocator.targetsWaiting, k)
+	}
```

Comment:
```suggestion
	allocator.targetsWaiting = make(map[string]TargetItem)
```

Don't think it's necessary to delete each entry individually, just replace the whole map.